crates/sol-macro/src/expand/contract.rs

```diff
@@ -314,7 +314,7 @@ impl<'a> CallLikeExpander<'a> {
                                 .map(Self::#variants)
                         })*
                         s => ::core::result::Result::Err(::alloy_sol_types::Error::unknown_selector(
-                            Self::NAME,
+                            <Self as ::alloy_sol_types::SolInterface>::NAME,
```

Comment:
Part of 1 of issue #361
`expected reference &'static str found enum constructor`